com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -53,13 +54,29 @@ internal static string PrefabDebugHelper(NetworkPrefab networkPrefab)
             return $"{nameof(NetworkPrefab)} \"{networkPrefab.Prefab.gameObject.name}\"";
         }
 
+        private InterestManager<NetworkObject> m_InterestManager;
+
+        // For unit (vs. integration) testing and for better decoupling, we don't want to have to require Initialize()
+        //  to use the InterestManager
+        internal InterestManager<NetworkObject> InterestManager
+        {
+            get
+            {
+                if (m_InterestManager == null)
+                {
+                    m_InterestManager = new InterestManager<NetworkObject>();
+                }
+                return m_InterestManager;
+            }
+        }
         internal SnapshotSystem SnapshotSystem { get; private set; }
         internal NetworkBehaviourUpdater BehaviourUpdater { get; private set; }
 
         internal MessagingSystem MessagingSystem { get; private set; }
 
         private NetworkPrefabHandler m_PrefabHandler;
 
+
```

Comment:
Is this needed ?

Reply:
As in, do we need the getter?  I ran into problems with testing and this lazy initialization scheme was my solution.  I actually am not a fan on how all the other services above (e.g. NetworkBehaviourUpdater) don't initialize until `NetworkManager::Initialize()` is called and am interested in porting them to this way, because that means to exercise them I *must* call `StartServer` / `StartHost`, and maybe I don't want to require a stood-up, connected NM to test / exercise it.

Reply:
I mean the extra space :P